### PR TITLE
Add Issues Generation as per RFC #0084

### PR DESCRIPTION
**Pull request by jromero**
### Summary

This PR introduces the following changes:

1. A GitHub Action that monitors PRs for the purpose of easier issue management:
    1. Creates an initial comment instructing maintainers that is updated as issues are queued.
    2. Parses comments for `/queue-issue` or `/unqueue-issue` to track issues to create.
2. A `node` based CLI to query and create queued issues (currently necessary to integrate with `merge-rfc.sh`)
3. Changes to `merge-rfc.sh` to query and create queued issues

### Files of interest

This PR looks a lot larger than it is because of the JavaScript GitHub Action configuration and boilerplate. Here are the files I would recommend reviewers focus on...

- [Action and CLI source code](https://github.com/jromero/issue-generation-action)
- [.github/workflows/issues-generation.yml](https://github.com/buildpacks/rfcs/compare/main...jromero:feature/issue-generation?expand=1#diff-170370bb8ddecaf674c7caaf7fd1fbe15614cbe05d54e82a59d993e989d931a1)
- [merge-rfc.sh](https://github.com/buildpacks/rfcs/compare/main...jromero:feature/issue-generation?expand=1#diff-8584301d886a21d71e0a7e0cd17b25aa5f99cb9eb4487ac0d0da6c6530818c04)

### Examples

- [RFC Pull Request](https://github.com/jromero/rfcs/pull/23)
- [Results after running `merge-rfc.sh`](https://github.com/jromero/rfcs/pull/26)

### Screenshots

image

### CODE REVIEW

**Comment by jromero** [comment]
@buildpacks/core-team - any feedback on this? 

As I think about it more, I do wonder if it would be better if the GH action lived in it's own repo as opposed to it being embedded here. Would that make it easier to get merged in? One downside would be maintenance since it would require making changes elsewhere in the future.

**Comment by jkutner** [comment]
@jromero sorry, I haven't had time to look at this in detail but at first glance it seems great. I **don't** think you should move it to another repo. We can merge it here without all the pomp and circumstance of an RFC PR.

File: merge-rfc.sh
Comment (jkutner) [inline merge-rfc.sh:1]: bash4life

File: merge-rfc.sh
Comment (jkutner) [inline merge-rfc.sh:36]: is this to accommodate forks?

File: .github/workflows/issues-generation.yml
Comment (jkutner) [inline .github/workflows/issues-generation.yml:23]: I think this could probably use `GITHUB_TOKEN`? Not sure if that's matters much.

**Review by jkutner — APPROVED** [review]
Approving, but my JS/TS is weak, so I'm not sure I would/could review that well. It might be good to get some eyes on this from someone who has stronger JS/TS, but I'm comfortable merging this without review from the full core team too.

File: merge-rfc.sh
Comment (jromero) [reply merge-rfc.sh:36]: Yes, essentially. Something that I needed as I was developing and didn't see a specific reason to revert it.

File: .github/workflows/issues-generation.yml
Comment (jromero) [reply .github/workflows/issues-generation.yml:23]: This needs to be a token for @buildpack-bot since it's that account we are making all the operations as.

File: .github/workflows/actions/issues-generation/dist/index.js
Comment (hone) [reply .github/workflows/actions/issues-generation/dist/index.js:9498]: ignore

File: .github/workflows/actions/issues-generation/.editorconfig
Comment (hone) [inline .github/workflows/actions/issues-generation/.editorconfig:1]: is this something we normally include?

File: .github/workflows/actions/issues-generation/README.md
Comment (hone) [inline .github/workflows/actions/issues-generation/README.md:55]: What's this `LEARNING_GITHUB_TOKEN`?

File: merge-rfc.sh
Comment (hone) [inline merge-rfc.sh:156]: What's the configured git user here for the `signoff` and `commit`?

**Comment by hone** [comment]
> @buildpacks/core-team - any feedback on this?
> 
> As I think about it more, I do wonder if it would be better if the GH action lived in it's own repo as opposed to it being embedded here. Would that make it easier to get merged in? One downside would be maintenance since it would require making changes elsewhere in the future.

I looked through the code and it looks good to me. I would prefer to see it moved out of the RFCs repo with just consuming the actions remaining here, but happy to do that in a separate PR.

**Review by hone — APPROVED** [review]

File: merge-rfc.sh
Comment (jromero) [reply merge-rfc.sh:156]: This would be the current user running `./merge-rfc.sh`. You, me, anyone. :)

**Comment by jromero** [comment]
> > @buildpacks/core-team - any feedback on this?
> > As I think about it more, I do wonder if it would be better if the GH action lived in it's own repo as opposed to it being embedded here. Would that make it easier to get merged in? One downside would be maintenance since it would require making changes elsewhere in the future.
> 
> I looked through the code and it looks good to me. I would prefer to see it moved out of the RFCs repo with just consuming the actions remaining here, but happy to do that in a separate PR.

PR has been updated to externalize [`issues-generation` action and CLI](https://github.com/jromero/issue-generation-action)

**Comment by jromero** [comment]
Update, feel free to play with it here: https://github.com/jromero/rfcs/pull/27

**Comment by hone** [comment]
> PR has been updated to externalize [`issues-generation` action and CLI](https://github.com/jromero/issue-generation-action)

Thanks this looks good to me @jromero.

**Review by jkutner — APPROVED** [review]

**Comment by jromero** [comment]
@nebhale / @ekcasey - a review of this would be particularly helpful given you both have worked on the `merge-rfc.sh` script. Plus I need one more review according to GH. :D

**Review by nebhale — APPROVED** [review]

**Comment by jromero** [comment]
@buildpacks/core-team what's the process for merging in functional changes to this repository? Can this be merged in it's current state?

**Pull request merged**
